experimental/packages/opentelemetry-sdk-node/src/sdk.ts

```diff
@@ -196,19 +197,29 @@ export class NodeSDK {
         { [SemanticResourceAttributes.SERVICE_NAME]: this._serviceName }
       ));
 
-    if (this._tracerProviderConfig) {
-      const tracerProvider = new NodeTracerProvider({
-        ...this._tracerProviderConfig.tracerConfig,
+    if (this._tracerProviderConfig || this._useEnvExporters) {
+      const Provider =
+        this._tracerProviderConfig ? NodeTracerProvider : TracerProviderWithEnvExporters;
+
+      const tracerProvider = new Provider ({
+        ...this._tracerProviderConfig?.tracerConfig,
         resource: this._resource,
       });
 
       this._tracerProvider = tracerProvider;
 
-      tracerProvider.addSpanProcessor(this._tracerProviderConfig.spanProcessor);
-      tracerProvider.register({
-        contextManager: this._tracerProviderConfig.contextManager,
-        propagator: this._tracerProviderConfig.textMapPropagator,
-      });
+      const processors = this.retrieveSpanProcessors(tracerProvider);
+
+      if (processors) {
+        processors.forEach((processor: SpanProcessor) => {
+          tracerProvider.addSpanProcessor(processor);
+        });
+
+        tracerProvider.register({
```

Comment:
I believe we can handle the span processor registration in the `TracerProviderWithEnvExporters.register` instead.

Reply:
@legendecas Can you clarify what you mean? `TracerProviderWithEnvExporters.register` is essentially called when `tracerProvider.register` is called on line 218 in the case when the user wants to setup exporters from env (when `this._useEnvExporters` is set to `true`) because a `TracerProviderWithEnvExporters` tracer provider is [selected here](https://github.com/open-telemetry/opentelemetry-js/blob/e02d73981813f0efec72642e7e3d4ee10b53d0c6/experimental/packages/opentelemetry-sdk-node/src/sdk.ts#L202).

Reply:
I could add the span processors right after configuring them in the `TracerProviderWithEnvExporters` constructor:

```js
 this.spanProcessors = this.configureSpanProcessors(this.configuredExporters);
  this.spanProcessors.forEach(processor => {
    this.addSpanProcessor(processor)
  });
  ```
  And then I won't need to use `retrieveSpanProcessors` method in the `NodeSDK` and I can update the logic [here](https://github.com/open-telemetry/opentelemetry-js/blob/e02d73981813f0efec72642e7e3d4ee10b53d0c6/experimental/packages/opentelemetry-sdk-node/src/sdk.ts#L211-L222) to this below which is similar to what the `NodeSDK` does before this pr:
  
  ```js
    if (this._tracerProviderConfig) {
      tracerProvider.addSpanProcessor(this._tracerProviderConfig.spanProcessor)
    }

    tracerProvider.register({
      contextManager: this._tracerProviderConfig?.contextManager,
      propagator: this._tracerProviderConfig?.textMapPropagator,
    });

Reply:
@svetlanabrennan I like the idea of adding them in the constructor. :slightly_smiling_face:  That this was not the case is also where my confusion in [this comment](https://github.com/open-telemetry/opentelemetry-js/pull/3143#discussion_r963814663) came from. 

It looked like it would be possible to just re-add the `SpanProcessor`s back to a "normal" `TracerProvider` without the need of a `TracerProviderWithEnvExporters` - but this change should take care of my concerns. :slightly_smiling_face:

Reply:
I agree with you that this pattern (with calling `addSpanProcessor`) should be handled in `TracerProviderWithEnvExporters` own constructor.